src/modules/75-cd/components/EnvironmentGroups/CreateEnvironmentGroupModal.tsx

```diff
@@ -0,0 +1,418 @@
+  fileName: `environmentGroup.yaml`,
+  entityType: 'EnvironmentGroup',
+  width: '100%',
+  height: 250,
+  showSnippetSection: false,
+  yamlSanityConfig: {
+    removeEmptyString: false,
+    removeEmptyObject: false,
+    removeEmptyArray: false
+  }
+}
+
+export default function CreateEnvironmentGroupModal({ closeModal }: CreateEnvironmentGroupModalProps): JSX.Element {
+  const { orgIdentifier, projectIdentifier, accountId, module } = useParams<ProjectPathProps & ModulePathParams>()
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const formikRef = useRef<FormikProps<EnvironmentGroupResponseDTO>>()
+  const loadMoreRef = useRef(null)
+  const pageSize = useRef(20)
+
+  const [yamlHandler, setYamlHandler] = useState<YamlBuilderHandlerBinding | undefined>()
+  const [selectedView, setSelectedView] = useState<SelectedView>(SelectedView.VISUAL)
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [createLoading, setCreateLoading] = useState(false)
+
+  const { getString } = useStrings()
+  const history = useHistory()
+  const { showSuccess, showError, clear } = useToaster()
+
+  useEffect(() => {
+    inputRef.current?.focus()
+  }, [])
+
+  const onSubmit = async (value: EnvironmentGroupResponseDTO) => {
+    setCreateLoading(true)
+    try {
+      const values = cleanData(value)
+      const body = yamlStringify(
+        JSON.parse(
+          JSON.stringify({
+            environmentGroup: { ...values }
+          })
+        )
+      )
+      const response = await createEnvironmentGroupPromise({
+        body,
+        queryParams: {
+          accountIdentifier: accountId,
+          orgIdentifier,
+          projectIdentifier
+        },
+        requestOptions: { headers: { 'Content-Type': 'application/yaml' } }
+      })
+
+      if (response.status === 'SUCCESS') {
+        clear()
+        showSuccess(getString('common.environmentGroup.created'))
+        history.push(
+          routes.toEnvironmentGroupDetails({
+            orgIdentifier,
+            projectIdentifier,
+            accountId,
+            module,
+            environmentGroupIdentifier: defaultTo(response.data?.envGroup?.identifier, ''),
+            sectionId: EnvironmentGroupDetailsTab.ENVIRONMENTS
+          })
+        )
+      } else {
+        throw response
+      }
+    } catch (e: any) {
+      showError(getErrorInfoFromErrorObject(e, true))
+    }
+    setCreateLoading(false)
+  }
+
+  const handleModeSwitch = useCallback(
+    (view: SelectedView) => {
+      if (view === SelectedView.VISUAL) {
+        const yaml = defaultTo(yamlHandler?.getLatestYaml(), '')
+        const yamlVisual = parse(yaml).environmentGroup as EnvironmentGroupResponseDTO
+
+        if (yamlVisual) {
+          formikRef.current?.setValues({
+            ...(cleanData(yamlVisual) as EnvironmentGroupResponseDTO)
+          })
+        }
+      }
+
+      setSelectedView(view)
+    },
+    [yamlHandler?.getLatestYaml]
```

Comment:
Codacy has a fix for the issue: [React Hook useCallback has a missing dependency: 'yamlHandler'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232281)
```suggestion
    [yamlHandler]
```